imports/api/transactions/statement-entries/methods.js

```diff
@@ -201,18 +201,34 @@ export const recognize = new ValidatedMethod({
         Log.debug('Matching bill:', matchingBill);
       }
     }
+    let partner;
+    const entryName = entry.nameOrType();
+    if (entryName) {
+      Log.debug('Looking for partner', entryName, 'in', entry.communityId);
+      const recognizedName = Recognitions.get('name', entryName, { communityId }) || entryName;
+      partner = Partners.findOne({ communityId: entry.communityId, 'idCard.name': recognizedName });
+    } else {
+      Log.debug('No partner on statement');
+    }
+    let contraBankAccount;
+    if (!partner) {
+      const recognizedContraBAN = (entry.contraBAN && Recognitions.get('BAN', entry.contraBAN, { communityId })) || entry.contraBAN;
+      contraBankAccount = recognizedContraBAN && Accounts.findOne({ communityId, category: 'bank', BAN: recognizedContraBAN });
+    }
     // ---------------------------
     // 0th grade - 'direct' match: We find an existing payment tx, which can be mathched to this entry
     // ---------------------------
-    if (community.settings.paymentsWoStatement) {
-      const matchingTxs = Transactions.find({ communityId, category: 'payment', valueDate: entry.valueDate, amount: Math.abs(entry.amount) }).fetch();
+    if (community.settings.paymentsWoStatement && !contraBankAccount) {
+      const matchingTxs = Transactions.find({ communityId, category: 'payment', valueDate: entry.valueDate, amount: Math.abs(entry.amount) })
+        .fetch().filter(t => !t.isReconciled());
       if (matchingTxs) {
         let matchingTx;
         let confidence;
         if (matchingBill) matchingTx = matchingTxs.find(tx => tx.bills?.[0]?.id === matchingBill._id);
         if (matchingTx) confidence = 'primary';
         else {
-          matchingTx = matchingTxs[0];
+          if (partner) matchingTx = matchingTxs.find(tx => tx.partnerId === partner._id);
```

Comment:
A find az egy cursor-t ad vissza. Ide nem findOne-t akartál irni?
Igazábol szerintem a legjobb az lenne, ha find.fetch lenne. Es ha egyelemű eredményt kapunk a confidence success, ha pedig tobb elemű eredmenyt kapunk, akkor lehetne a confidence warning. Mert ilyenkor van az amikor tobb talalat kozul nem tudtuk eldonteni melyikhez kell parositani. Jo lenne ha szemmel is ránézne a könyvelést végző.

Reply:
Az egy array.find, ami csak az első megfelelő elemet adja vissza. Legyen egy filter inkább? És ha több elemű eredményt kapunk, akkor ne legyen match se? Vagy egyet válasszunk ki azért, de warning a confidence?

Reply:
Ja jo. Kicsit elore gondolkodtam. Mert a masik branchen mar ugye az keszul, hogy a reconciled: false beteheto legyen a selectorba, nem is fog kelleni fetch.filter oda.
Szerintem ha egy elemet talalunk az egy success vagy info szintu match. De ha tobb elem is megfelel, akkor warning legyen a confidence. Mindenkeppen nezzen ra, hogy jo e az amit random kiboktuk az egyiket.